stable/database/files/nuobackup

```diff
@@ -87,8 +87,18 @@ function execute_backup() {
                --key "$NUODB_BACKUP_KEY/$db_name/${backup_group}/failed" \
                --value "${failed} ${backupset}" --expected-value "${failed}"
          fi
-      fi 
-      exit $retval
+      fi
+
+      if [ "$type" = "journal" ] && echo "$output" | grep -q "Full or incremental hot copy must be completed before journal hot copy"; then
+         echo "Executing incremental hot copy as journal hot copy is temporarily suspended"
+         execute_backup "incremental" || return $?
+         # retry the journal backup which should suceed now
+         execute_backup "journal"
```

Comment:
This function is executing itself. Is there a risk that we keep issuing journal backups forever because `execute_backup "incremental"` did not return the correct code to signal that it failed? I would set a variable to prevent more than one "nested" invocation of `execute_backup "journal"` being generated.